doc/en/monkeypatch.rst

```diff
@@ -8,46 +8,222 @@ Sometimes tests need to invoke functionality which depends
+        # for this test with the monkeypatch.
         x = getssh()
-        assert x == '/abc/.ssh'
+        assert x == "/abc/.ssh"
+
+
+Monkeypatching returned objects: building mock classes
+------------------------------------------------------
+
+:py:meth:`monkeypatch.setattr` can be used in conjunction with classes to mock returned
+objects from functions instead of values.
+Imagine a simple function to take an API url and return the json response.
+
+.. code-block:: python
+
+    # Contents of app.py, a simple API retrieval example.
+    import requests
+
+
+    def get_json(url):
+        """Takes a URL, and returns the JSON."""
+        r = requests.get(url)
+        return r.json()
+
+We need to mock ``r``, the returned response object for testing purposes.
+The mock of ``r`` needs a ``.json()`` method which returns a dictionary.
+This can be done in our test file by defining a class to represent ``r``.
+
+.. code-block:: python
+
+    # Contents of test_app.py, a simple test for our API retrieval.
+    # Import requests for the purposes of monkeypatching.
+    import requests
+
+    # Our app.py that includes the get_json() function.
+    # This is the previous code block example.
+    import app
+
+    # A Custom class to be the mock return value.
+    # It needs to override the requests.Response returned from requests.get.
+    class MockResponse:
+
+        # The class has a json() method that we want to mock.
+        # The mock method always returns a specific testing dictionary.
+        @staticmethod
+        def json():
+            return {"mock_key": "mock_response"}
+
+
+    def test_get_json(monkeypatch):
+
+        # This function is used in the monkeypatch of requests.get().
+        # Any arguments may be passed and it will always return our
+        # mocked object, which only has the .json() method.
+        def mock_get(*args, **kwargs):
+            return MockResponse()
+
+        # Apply the monkeypatch for requests.get to mock_get.
+        monkeypatch.setattr(requests, "get", mock_get)
+
+        # Our simple but illustrative test.
+        # app.get_json, which contains requests.get, uses the monkeypatch.
+        # Therefore, our result is our mock.
+        result = app.get_json("https://fakeurl")
+        assert result["mock_key"] == "mock_response"
+
+
+``Monkeypatch`` applies the mock for ``requests.get`` with our ``mock_get`` function.
+The ``mock_get`` function returns an instance of the ``MockResponse`` class, which
+has a ``json()`` method defined to return a known testing dictionary and does not
+require any outside API connection.
+
+You can build the ``MockResponse`` class with the appropriate degree of complexity for
+the scenario you are testing. For instance, it could include an ``ok`` property that
+always returns ``True``, or return different values from the ``json()`` mocked method
+based on input strings.
+
+This mock can be shared across tests using the ``fixture`` syntax:
```

Comment:
```suggestion
This mock can be shared across tests using a ``fixture``:
```

A fixture is not really a new syntax. 😁

Reply:
Applied in https://github.com/pytest-dev/pytest/commit/7fa0e0405d4c051d176ba40ae4205dc3393e27a7